stream/src/main/scala/org/apache/pekko/stream/scaladsl/Flow.scala

```diff
@@ -1527,6 +1527,28 @@ trait FlowOps[+Out, +Mat] {
    */
   def dropWhile(p: Out => Boolean): Repr[Out] = via(DropWhile(p))
 
+  /**
+   * Emit true and complete the stream as soon as there's an element satisfy the predicate.
+   * Or emit false when upstream complete and no element satisfied the predicate.
+   *
+   * The stream will produce true for empty stream (i.e. completes before signalling any elements).
+   *
+   * The stream will never complete if there's no element satisfied the predicate and upstream isn't completed.
+   *
+   * Adheres to the [[ActorAttributes.SupervisionStrategy]] attribute.
+   *
+   * '''Emits when''' predicate returned true or upstream completes
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' predicate returned true or upstream completes
+   *
+   * '''Cancels when''' predicate returned true or downstream cancels
+   */
+  def exists(p: Out => Boolean): Repr[Boolean] = map(p).takeWhile(!_, inclusive = true).fold(true) {
+    case (_, incoming) => incoming
```

Comment:
need a `name("exists")` for this subFlow.

Reply:
in the Flow/Source, the name attributes should placed in `DefaultAttributes`.
And you will need via( new  Fold...withAttributes(...)`.